rectools/metrics/intersection.py

```diff
@@ -71,24 +70,15 @@ def calc_per_user(self, reco: pd.DataFrame, ref_reco: pd.DataFrame) -> pd.Series
         pd.Series:
             Values of metric (index - user id, values - metric value for every user).
         """
-        self._check(reco)
-        assert set(ref_reco.columns) >= {Columns.User, Columns.Item, Columns.Rank}
+        self._check(reco, ref_reco=ref_reco)
 
         if ref_reco.shape[0] == 0:
             return pd.Series(index=pd.Series(name=Columns.User, dtype=int), dtype=np.float64)
 
-        if ref_reco is reco:
```

Comment:
Are you sure we should drop this? In `cross_validate` we will get boost to speed without actually calculating recall because we are passing the same object for both `reco` and `ref_reco`

Reply:
I totally forgot about the `cross_validate`, thanks. Refactored it as well

Returned this optimisation